### PR TITLE
docs($animate): include docs for `on`, `off`, `enabled` and `cancel`

### PR DESCRIPTION
**Pull request by matsko**

### CODE REVIEW

File: src/ng/animate.js [hunk 1]
```diff
@@ -270,10 +270,99 @@ var $AnimateProvider = ['$provide', function($provide) {
...
       // we don't call it directly since non-existant arguments may
       // be interpreted as null within the sub enabled function
+
+      /**
+       *
+       * @ngdoc method
+       * @name $animate#on
+       * @kind function
+       * @description Sets up an event listener to fire whenever the animation event (enter, leave, move, etc...)
+       *    has fired on the given element or amount any of its children. Once the listener is fired, the provided callback
```
Comment (gkalpak) [inline src/ng/animate.js]: I don't get the "amount" part here (but maybe it's just my English).

Comment (matsko) [inline src/ng/animate.js]: Yes that is a mistake on my end. I too learned English long ago I forget these things :P

File: src/ng/animate.js [hunk 2]
```diff
@@ -270,10 +270,99 @@ var $AnimateProvider = ['$provide', function($provide) {
...
+       * $animate.on('enter', container,
+       *    function(element, phase) {
+       *      // cool we detected an enter animation within the container
+       *    }
+       * );
+       * ```
+       *
+       * @param {string} event the animation event that will be captured (e.g. enter, leave, move, addClass, removeClass, etc...)
+       * @param {DOMElement} container the container element that will capture each of the animation events that are fired on itself
+       *     as well as well as amount its children
```
Comment (gkalpak) [inline src/ng/animate.js]: _as well as well_  -->  _as well_

(And I still don't get the "amount" part :))

File: src/ng/animate.js [hunk 3]
```diff
@@ -270,10 +270,99 @@ var $AnimateProvider = ['$provide', function($provide) {
...
+       *
+       * @return {boolean} whether or not animations are enabled
+       */
       enabled: $$animateQueue.enabled,
 
+      /**
+       * @ngdoc method
+       * @name $animate#cancel
+       * @kind function
+       * @description Canels the provided animation.
```
Comment (gkalpak) [inline src/ng/animate.js]: _Canels_  -->  _Cancels_

--
BTW, a couple of lines below, is `runner.cancel && runner.end();` a mistake ?

File: src/ng/animate.js [hunk 4]
```diff
@@ -270,10 +270,99 @@ var $AnimateProvider = ['$provide', function($provide) {
...
+       * @kind function
+       * @description Deregisters an event listener based on the event which has been associated with the provided element. This method
+       * can be used in three different ways depending on the arguments:
+       *
+       * ```js
+       * // remove all the animation event listeners listening for `enter`
+       * $animate.off('enter');
+       *
+       * // remove all the animation event listeners listening for `enter` on the given element and its children
+       * $animate.off('enter', element);
```
Comment (gkalpak) [inline src/ng/animate.js]: I would probably call it `container` to be consistent with param description.

File: src/ng/animate.js [hunk 5]
```diff
@@ -270,10 +270,99 @@ var $AnimateProvider = ['$provide', function($provide) {
...
+       *
+       * // remove the event listener function provided by `listenerFn` that is set to listen on `enter` for the given `element`
+       * $animate.off('enter', element, listenerFn);
```
Comment (gkalpak) [inline src/ng/animate.js]: I would probably call it `callback` to be consistent with param description.

File: src/ng/animate.js [hunk 6]
```diff
@@ -270,10 +270,99 @@ var $AnimateProvider = ['$provide', function($provide) {
...
+       * // returns true or false if animations are enabled for an element
+       * $animate.enabled(element);
+       *
+       * // changes the enabled state for an element and its children
+       * $animate.enabled(element, true);
+       * $animate.enabled(element, false);
+       * ```
+       *
+       * @param {DOMElement} element the element that will be considered for checking/setting the enabled state
+       * @param {boolean} enabled whether or not the animations will be enabled for the element
```
Comment (gkalpak) [inline src/ng/animate.js]: Both parameters should be marked as optional.

File: src/ng/animate.js [hunk 7]
```diff
@@ -270,10 +270,99 @@ var $AnimateProvider = ['$provide', function($provide) {
...
+       * can be used in three different ways depending on the arguments:
+       *
+       * ```js
+       * // remove all the animation event listeners listening for `enter`
+       * $animate.off('enter');
+       *
+       * // remove all the animation event listeners listening for `enter` on the given element and its children
+       * $animate.off('enter', element);
+       *
+       * // remove the event listener function provided by `listenerFn` that is set to listen on `enter` for the given `element`
```
Comment (gkalpak) [inline src/ng/animate.js]: I would probably change it to "...listen for `enter` on the given element and its children" (to be consistent with the phrasing above ).

**Comment by petebacondarwin** [comment]
@matsko - Looks like @gkalpak has given this a good seeing to. Feel free to merge once you've dealt this his comments.

File: src/ng/animate.js (hunk 1 above)
Comment (matsko) [inline src/ng/animate.js]: I had meant to say `among`.

File: src/ng/animate.js (hunk 2 above)
Comment (matsko) [inline src/ng/animate.js]: same thing here.

File: src/ng/animate.js (hunk 3 above)
Comment (matsko) [inline src/ng/animate.js]: Yes this was a mistake, but it didn't cause any bugs since `cancel` is a valid method on the runner.

Comment (matsko) [inline src/ng/animate.js]: Fixed.

**Comment by matsko** [comment]
Merged in as https://github.com/angular/angular.js/commit/4d437ba9f7e48b1b2c1ef00958510cc8f1c093b1. Thank you guys.

**Pull request closed without merging**